### PR TITLE
feat: pcd loader support for compressed binary data

### PR DESCRIPTION
**Pull request by xhy279**
We got a situation to parse compressed binary PCD data and this should work.

### CODE REVIEW

File: modules/pcd/src/lib/parse-pcd.ts [hunk 1]
```diff
@@ -312,3 +315,122 @@ function parsePCDBinary(pcdHeader: PCDHeader, data: ArrayBufferLike): HeaderAttr
...
+  }
+
+  return {
+    position,
+    normal,
+    color
+  };
+}
+
+function decompressLZF(inData, outLength) {
```
Comment (ibgreen) [inline modules/pcd/src/lib/parse-pcd.ts:394]: Please 
- add JSDoc header describing what the function does (link to format spec if possible)
- add typescript types.

Also I assume this is copied from some existing repo, if so can we add a link and mention license and provide attribution? Possibly put this util in a separate file?

**Review by ibgreen — APPROVED** [review]
Looks like a great contribution, thanks. 

Will land if you can address the comments.

**Review by ibgreen — APPROVED** [review]
Would also be nice to make minimal updates in 
- docs/whatsnew.md, mention the new capability.
- modules/pcd/docs/... any mention here would be good.

Also we really need a test file, without tests this will be hard to maintainl

File: modules/pcd/src/lib/parse-pcd.ts (hunk 1 above)
Comment (xhy279) [reply modules/pcd/src/lib/parse-pcd.ts]: added docs and types please review it  again

Comment (xhy279) [reply modules/pcd/src/lib/parse-pcd.ts]: > Please
> 
> * add JSDoc header describing what the function does (link to format spec if possible)
> * add typescript types.
> 
> Also I assume this is copied from some existing repo, if so can we add a link and mention license and provide attribution? Possibly put this util in a separate file?

added docs and types please review it again.

File: docs/whats-new.md [hunk 2]
```diff
@@ -59,6 +59,9 @@ Target Release Date: Nov 2021.
 
 - New experimental [`ParquetLoader`](modules/parquet/docs/api-reference/parquet-loader)
 
+**@loaders.gl/pcd**
+- `PCDLoader` now supports for compressed binary data.
```
Comment (ibgreen) [inline docs/whats-new.md:63]:
```suggestion
- `PCDLoader` now supports compressed binary data.
```

File: modules/pcd/docs/api-reference/pcd-loader.md [hunk 3]
```diff
@@ -12,7 +12,7 @@ The `PCDLoader` loads point cloud in the Point Cloud Data (PCD) format.
 | Worker Thread Support | Yes                                                                                    |
 | Streaming Support     | No                                                                                     |
 
-Note: Currently only `ascii` and `binary` subformats are supported. Compressed binary files are currently not supported.
+Note: Currently supports for `ascii`, `binary` and compressed binary files.
```
Comment (ibgreen) [inline modules/pcd/docs/api-reference/pcd-loader.md:15]:
```suggestion
Note: Currently supports `ascii`, `binary` and compressed binary files. 
```

File: modules/pcd/src/lib/decompress-lzf.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
+/* eslint-disable */
+/**
+ * from https://gitlab.com/taketwo/three-pcd-loader/blob/master/decompress-lzf.js
+ * @param inData
+ * @param outLength
+ * @returns
+ */
+export function decompressLZF(inData, outLength) {
```
Comment (ibgreen) [inline modules/pcd/src/lib/decompress-lzf.ts:8]:
```suggestion
export function decompressLZF(inData: Uint8Array, outLength: number): Uint8Array {
```

**Review by ibgreen — APPROVED** [review]

**Review by ibgreen — COMMENTED** [review]
@xhy279 I will land this but I am uncomfortable with not having a test file. Do you have any small sample file in binary_compressed format that could be shared?

**Pull request merged**